nnsmith/materialize/torch/forward.py

```diff
@@ -19,6 +23,36 @@
 # forward_fn:  forward
 
 
+@operator_impl(ConcreteOp)
+def forward_fn(op: ConcreteOp):
+    expr: str = op.target_str
+    offset = 0
+    if expr.startswith("torch.Tensor."):
+        expr = f'tensors[0].{expr[len("torch.Tensor."):]}'
+        offset = 1
+    target = eval(expr)
+    args_flatten, args_treespec = pytree.tree_flatten(op.args[offset:])
+    kwargs_flatten, kwargs_treespec = pytree.tree_flatten(op.kwargs)
+
+    def inner(*tensors: List[torch.Tensor]):
+        i_ts = offset
+        for i, v in enumerate(args_flatten):
+            if v is ConcreteOp.empty:
+                args_flatten[i] = tensors[i_ts]
+                i_ts += 1
+        for i, v in enumerate(kwargs_flatten):
+            if v is ConcreteOp.empty:
+                kwargs_flatten[i] = tensors[i_ts]
+                i_ts += 1
+        args = pytree.tree_unflatten(args_flatten, args_treespec)
+        kwargs = pytree.tree_unflatten(kwargs_flatten, kwargs_treespec)
+        return target(*args, **kwargs)
+
+    setattr(inner, "_target", target)
```

Comment:
Not sure if we can simplify the hack. Putting extra attributes to a local function looks a bit odd...

Reply:
Maybe it's okay to get back to returning a pair (i.e., `Tuple[Callable, Optional[nn.Module]]`)